internal/k8shandler/reconciler.go

```diff
@@ -103,6 +103,12 @@ func Reconcile(cl *logging.ClusterLogging, requestClient client.Client, reader c
 			return clusterLoggingRequest.Cluster, fmt.Errorf("unable to create or update logstore for %q: %v", clusterLoggingRequest.Cluster.Name, err)
 		}
 
+		// Migrate visualization spec for Kibana
```

Comment:
I don't see why we don't have something like https://github.com/openshift/cluster-logging-operator/blob/a9b7a1c4903704e0e073a6a61f8108f9a8606e9e/internal/migrations/migrate_clusterlogforwarder.go#L10 for ClusterLogging which migrates the collector (and visualization) at the controller level which is where I would have expected us to hook into migrating visualizatin

Reply:
Actually looking into the migrations a bit more, there is a `migrate_collection.go` file that migrates ClusterLogging: https://github.com/openshift/cluster-logging-operator/blob/a9b7a1c4903704e0e073a6a61f8108f9a8606e9e/internal/migrations/migrate_collection.go#L10

I can move migrating visualization into there to reduce files and code.

This migration is still in the `k8shandler/reconciler.go`:
https://github.com/openshift/cluster-logging-operator/blob/a9b7a1c4903704e0e073a6a61f8108f9a8606e9e/internal/k8shandler/reconciler.go#L252

I don't see why we couldn't move it to the controller and wouldn't be opposed to the refactor.

Reply:
migrate should be really early in the reconciliation which is what I hope we did for CLF.  and should actually introduce a new function: 
```
clusterLogging.Spec = migrations.MigrateClusterLogging
```

I think they exist where they do now because we do "gets" on the resources from one controller or the other which makes it difficult to locate that code only at the top of one controller

Reply:
I think this should be captured in another story as it is deviating away from the heart of this specific issue. I propose to keep these changes as is and work on refactoring in another Jira story. What do you think?

Reply:
In the interest of "minimizing" changes... lets simply introduce a "migrateClusterLoggingSpec" and encapsulate the collection and visualization in that .... and replace the places where we directly call "migrateCollectionSpec" with the new function.  This will suffice for me until we can otherwise refactor

Reply:
Refactored `migrate_collection.go` to `migrate_clusterlogging.go` and encapsulated migrating visualization within it.